converters/fromZigbee.js

```diff
@@ -1477,6 +1477,30 @@ const converters = {
             };
         },
     },
+    develco_ias_enroll: {
+        cluster: 'ssIasZone',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const zoneState = msg.data.zoneState;
+            const iasCieAddr = msg.data.iasCieAddr;
+            const zoneId = msg.data.zoneId;
+
+            return {
+                enrolled: zoneState,
+                IASCIEAddress: iasCieAddr,
+                zoneID: zoneId,
+            };
+        },
+    },
+    develco_duration: {
+        cluster: 'ssIasWd',
+        type: ['attributeReport', 'readResponse'],
+        convert: (model, msg, publish, options, meta) => {
+            const result = {};
+            if (msg.data.hasOwnProperty('maxDuration')) result['maxDuration'] = msg.data.maxDuration;
```

Comment:
`maxDuration` -> `max_duration`

Reply:
@Koenkk 

I really struggle with this.
I've done the change but I experience and issue of getting the value of `max_duration`. I get it when I do a re-configure from the UI, and I get it by pressing the update from exposes. As soon as there are a new `attributeReport` or a command, the value return to `null`. I can't figure out what I'm missing.

Can you please assist by having a look at the code below?

```
const fzLocal = {

    smszb120_fw: {
        cluster: 'genBasic',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            const result = {};
            if (0x8000 in msg.data) {
                result.current_firmware = msg.data[0x8000].join('.');
            }
            return result;
        },
    },

    ias_smoke_alarm_1_develco: {
        cluster: 'ssIasZone',
        type: ['commandStatusChangeNotification'],
        convert: (model, msg, publish, options, meta) => {
            const zoneStatus = msg.data.zonestatus;
            return {
                smoke: (zoneStatus & 1) > 0,
                battery_low: (zoneStatus & 1<<3) > 0,
                supervision_reports: (zoneStatus & 1<<4) > 0,
                restore_reports: (zoneStatus & 1<<5) > 0,
                test: (zoneStatus & 1<<8) > 0,
            };
        },
    },

    develco_ias_enroll: {
        cluster: 'ssIasZone',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            const zoneState = msg.data.zoneState;
            const iasCieAddr = msg.data.iasCieAddr;
            const zoneId = msg.data.zoneId;

            return {
                enrolled: zoneState,
                iascie_address: iasCieAddr,
                zone_id: zoneId,
            };
        },
    },

    max_duration: {
        cluster: 'ssIasWd',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            const result = {};
            if (msg.data.hasOwnProperty('maxDuration')) result['max_duration'] = msg.data.maxDuration;
            return result;  
        },
    },
}

const tzLocal = {

    develco_duration: {
        key: ['max_duration'],
        convertSet: async (entity, key, value, meta) => {
            await entity.write('ssIasWd', {'maxDuration': value});
            return {state: {max_duration: value}};
        },
        convertGet: async (entity, key, meta) => {
            await entity.read('ssIasWd', ['maxDuration']);
        },
    },

    develco_alarm: {
        key: ['alarm'],
        convertSet: async (entity, key, value, meta) => {
            const alarmState = (value === 'alarm' || value === 'OFF' ? 0 : 1);
            const info = (3 << 4) + ((alarmState) << 2);
            const values = {
                duration: value.hasOwnProperty('maxDuration') ? value.duration : 300,
            };
            await entity.command(
                'ssIasWd',
                'startWarning',
                {startwarninginfo: info, warningduration: values.duration},
                utils.getOptions(meta.mapped, entity),
            );
        },
    },
}

module.exports = [
    {
        zigbeeModel: ['SMSZB-120'],
        model: 'SMSZB-120',
        vendor: 'Develco',
        description: 'Smoke detector with siren',
        fromZigbee: [fz.temperature, fz.battery, fz.ias_smoke_alarm_1_develco, fz.ignore_basic_report, 
            fz.smszb120_fw, fzLocal.develco_ias_enroll, fzLocal.max_duration],
        toZigbee: [tzLocal.develco_duration, tzLocal.develco_alarm],
        ota: ota.zigbeeOTA,
        meta: {battery: {voltageToPercentage: '3V_2100'}},
        configure: async (device, coordinatorEndpoint, logger) => {
            const options = {manufacturerCode: 4117};
            const endpoint = device.getEndpoint(35);
            
            await reporting.bind(endpoint, coordinatorEndpoint, ['genPowerCfg', 'ssIasZone', 'ssIasWd', 'genBasic']);
            await reporting.batteryVoltage(endpoint);
            await endpoint.read('genBasic', [0x8000], options);
            await endpoint.read('ssIasZone', ['iasCieAddr', 'zoneState', 'zoneId']);
            await endpoint.read('ssIasWd', ['maxDuration']);

            const endpoint2 = device.getEndpoint(38);
            await reporting.bind(endpoint2, coordinatorEndpoint, ['msTemperatureMeasurement']);
            await reporting.temperature(endpoint2, {min: constants.repInterval.MINUTE, max: constants.repInterval.MINUTES_10, change: 10});
        },
        endpoint: (device) => {
            return {default: 35};
        },
        exposes: [e.temperature(), e.battery(), e.smoke(), e.battery_low(), e.test(),
            exposes.numeric('max_duration', ea.ALL).withUnit('s').withValueMin(0).withValueMax(600).withDescription('Duration of Siren'),
            exposes.binary('alarm', ea.SET, 'ON', 'OFF').withDescription('Manual Start of Siren')],
    },
];
```

Reply:
Can you provide the debug log of this?

Reply:
Sure...

Value are now `null`, and I go to exposes and press the **update**. It reads the value and update the `max_duration` attribute. Soon after, the device sends an `attributeReport` for `msTemperatureMeasurement` and the `max_duration` becomes `null` again.

```
Zigbee2MQTT:debug 22-07-2021 19:39:56: Publishing get 'get' 'max_duration' to '0x0015bc0031004b67'

Zigbee2MQTT:debug 22-07-2021 19:40:00: Received Zigbee message from '0x0015bc0031004b67', type 'readResponse', cluster 'ssIasWd', data '{"maxDuration":10}' from endpoint 35 with groupID 0

Zigbee2MQTT:info  22-07-2021 19:40:00: MQTT publish: topic 'zigbee2mqtt/0x0015bc0031004b67', payload '{"battery":100,"battery_low":false,"current_firmware":"4.0.2","enrolled":1,"iascie_address":"0x00124b0018e26fb0","linkquality":47,"max_duration":10,"restore_reports":true,"smoke":false,"supervision_reports":true,"temperature":22.37,"test":false,"update":{"state":"idle"},"update_available":false,"voltage":3000,"zone_id":23}'

Zigbee2MQTT:debug 22-07-2021 19:40:09: Received Zigbee message from '0x0015bc0031004b67', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":2225}' from endpoint 38 with groupID 0
Zigbee2MQTT:info  22-07-2021 19:40:09: MQTT publish: topic 'zigbee2mqtt/0x0015bc0031004b67', payload '{"battery":100,"battery_low":false,"current_firmware":"4.0.2","enrolled":1,"iascie_address":"0x00124b0018e26fb0","linkquality":44,"max_duration":null,"restore_reports":true,"smoke":false,"supervision_reports":true,"temperature":22.25,"test":false,"update":{"state":"idle"},"update_available":false,"voltage":3000,"zone_id":23}'
```

Reply:
This is really strange indeed, could you provide the herdsman debug log without leaving anything out starting from the `Publishing get 'get' 'max_duration'` until the `max_duration: null` is published?

Please post the log on pastebin.com and link it here.

Reply:
Here is the [full herdsman debug log](https://pastebin.com/p0XHvetw)

**FYI:**
1) I've added extra console logging as you see from the log.

fzLocal
```
    max_duration: {
        cluster: 'ssIasWd',
        type: ['attributeReport', 'readResponse'],
        convert: (model, msg, publish, options, meta) => {
            const result = {};
            if (msg.data.hasOwnProperty('maxDuration')) result['max_duration'] = msg.data.maxDuration;
                console.log('--------------------');
                console.log('Read: FZ.develco_duration');
                console.log('max_duration: ',result.max_duration);
                console.log('--------------------');
            return result;  
        },
    },
```

tzLocal
```
    max_duration: {
        key: ['max_duration'],
        convertSet: async (entity, key, value, meta) => {
            await entity.write('ssIasWd', {'maxDuration': value});
            return {state: {max_duration: value}};
        },
        convertGet: async (entity, key, meta) => {
            await entity.read('ssIasWd', ['maxDuration']);
            console.log('--------------------');
            console.log('READ: convertGet: TZ.develco_duration');
            console.log('--------------------');
        },
    },
```

2) I'm also extending the integration of a siren. It also supports both ssIasZone and ssIasWd. Exact same issue there. So this is not an issue with the device, there is either something in my code, or ... I don't know. 


The weird part are that if I KEEP the naming as `maxDuration`, it never returns to `null`. Only happens when I return it as `max_duration`.

Reply:
issue should be fixed in the latest-dev now.

Reply:
@Koenkk 

Can confirm the issue are resolved. I will re-write two of the converters as I see the `tz.warning` doesn't work at all as intended on several devices it are used. Expect the updated PR sometime tomorrow :)

Thanks for the fix Koen 🥇